sklearn/linear_model/_coordinate_descent.py

```diff
@@ -146,47 +149,35 @@ def _alpha_grid(
             "your estimator with the appropriate `alphas=` "
             "argument."
         )
-    n_samples = len(y)
```

Comment:
Could you keep `n_samples` and use it instead of `X.shape[0]`? Maybe even
```python
n_samples = X.shape[0]
```

Reply:
Done.